hack/update-staging-godeps.sh

```diff
@@ -23,102 +23,84 @@ set -o errexit
-echo "working in ${TARGET_DIR}"
-
-SKIP_RESTORE=${SKIP_RESTORE:-}
-if [ "${SKIP_RESTORE}" != "true" ]; then
-	echo "starting godep restore"
-	mkdir -p "${godepRestoreDir}"
-
-	# add the vendor folder so that we don't redownload things during restore
-	GOPATH="${godepRestoreDir}:${KUBE_ROOT}/staging:${ORIGINAL_GOPATH}"
-	# restore from kubernetes godeps to ensure we get the correct levels
-	# you get errors about the staging repos not using a known version control system
-	${godep} restore > ${godepRestoreDir}/godep-restore.log
-	echo "finished godep restore"
+echo "Checking whether godeps are restored"
+if ! kube::util::godep_restored 2>&1 | sed 's/^/  /'; then
+  echo -e '\nRun 'godep restore' to download dependencies.' 1>&2
+  exit 1
 fi
 
-echo "forcing fake godep folders to match the current state of master in tmp"
-for stagingRepo in $(ls ${KUBE_ROOT}/staging/src/k8s.io); do
-	echo "    creating ${stagingRepo}"
-	rm -rf ${godepRestoreDir}/src/k8s.io/${stagingRepo}
-	cp -a ${KUBE_ROOT}/staging/src/k8s.io/${stagingRepo} ${godepRestoreDir}/src/k8s.io
-
-	# we need a git commit in the godep folder, otherwise godep won't save
-	pushd ${godepRestoreDir}/src/k8s.io/${stagingRepo}
-	git init > /dev/null
-	# we need this so later commands work, but nothing should ever actually include these commits
-	# these are local only, not global
-	git config user.email "you@example.com"
-	git config user.name "Your Name"
-	git add . > /dev/null
-	git commit -qm "fake commit"
-	popd
-done
+kube::util::ensure_clean_working_dir
+kube::util::ensure-temp-dir
+kube::util::ensure_godep_version v74
+
+TMP_GOPATH="${KUBE_TEMP}/go"
 
 function updateGodepManifest() {
-	local repo=${1}
-
-	echo "starting ${repo} save"
-	mkdir -p ${TARGET_DIR}/src/k8s.io
-	# if target_dir isn't the same as source dir, you need copy
-	test "${KUBE_ROOT}/staging" = "${TARGET_DIR}" || cp -a ${KUBE_ROOT}/staging/src/${repo} ${TARGET_DIR}/src/k8s.io
-	# remove the current Godeps.json so we always rebuild it
-	rm -rf ${TARGET_DIR}/src/${repo}/Godeps
-	GOPATH="${godepRestoreDir}:${TARGET_DIR}"
-	pushd ${TARGET_DIR}/src/${repo}
-	${godep} save ./...
-
-	# now remove all the go files.	We'll re-run a restore, go get, godep save cycle in the sync scripts
-	# to get the commits for other staging k8s.io repos anyway, so we don't need the added files
-	rm -rf vendor
-
-	echo "rewriting Godeps.json to remove commits that don't really exist because we haven't pushed the prereqs yet"
-	GOPATH="${ORIGINAL_GOPATH}"
-	go run "${KUBE_ROOT}/staging/godeps-json-updater.go" --godeps-file="${TARGET_DIR}/src/${repo}/Godeps/Godeps.json" --client-go-import-path="${repo}"
-
-	popd
-	echo "finished ${repo} save"
+  local repo="${1}"
+  pushd "${TMP_GOPATH}/src/k8s.io/${repo}" >/dev/null
+    echo "Updating godeps for k8s.io/${repo}"
+    rm -rf Godeps # remove the current Godeps.json so we always rebuild it
+    GOPATH="${TMP_GOPATH}:${GOPATH}:${KUBE_ROOT}/staging" godep save ${V} ./... 2>&1 | sed 's/^/  /'
```

Comment:
Probably it doesn't hurt to have another copy in GOPATH, because the publish robot will rewrite the revision later. If that's the case, can we remove https://github.com/kubernetes/kubernetes/pull/41987/files#diff-79c7edfbbe6b22642e05ac8c9cd431e7R124?